superset/key_value/utils.py

```diff
@@ -61,9 +61,9 @@ def decode_permalink_id(key: str, salt: str) -> int:
 
 
 def get_uuid_namespace(seed: str) -> UUID:
-    md5_obj = md5()
-    md5_obj.update(seed.encode("utf-8"))
-    return UUID(md5_obj.hexdigest())
+    rd = random.Random()
```

Comment:
Won't this break existing namespaces? It seems that this logic was intended to keep UUIDs deterministic based on some input

Reply:
Yes, at first I thought the uuid was only used in cache, looks like that hash is stored in the db. To address your concern about deterministic I've replaced random with sha256. Will look to see how we can make this backwards compatible.

Reply:
I think you would need to make a migration that updates all existing entries from the MD5 based UUIDs to using SHA256 as the base. I think it should be doable, but we do need to be very careful with this. Ping @michael-s-molina , do you have any concerns here?